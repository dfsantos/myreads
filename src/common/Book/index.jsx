import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  title: PropTypes.string.isRequired,
  onChangeBookCategory: PropTypes.func.isRequired,
  coverLink: PropTypes.string.isRequired,
};

function Book({ title, coverLink, onChangeBookCategory }) {
  return (
    <div className="book">
      <div className="book-top">
        <div
          className="book-cover"
          style={{ width: 128, height: 193, backgroundImage: `url(${coverLink})` }}
        />
        <div className="book-shelf-changer">
          <select onChange={event => onChangeBookCategory(event.target.value)} defaultValue="none">
            <option value="none" disabled>
              Move to...
            </option>
            <option value="currentlyReading">Currently Reading</option>
            <option value="wantToRead">Want to Read</option>
            <option value="read">Read</option>
            <option value="none">None</option>
          </select>
        </div>
      </div>
      <div className="book-title">{title}</div>
      <div className="book-authors">{[].join(', ')}</div>
    </div>
  );
}

Book.propTypes = propTypes;

export default Book;
