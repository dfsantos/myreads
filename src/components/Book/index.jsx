import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Card, CardText, CardActions, CardMedia } from 'material-ui/Card';
import DropDownMenu from 'material-ui/DropDownMenu';
import MenuItem from 'material-ui/MenuItem';

const propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  description: PropTypes.string,
  authors: PropTypes.arrayOf(PropTypes.string),
  coverLink: PropTypes.string.isRequired,
  onChangeBookShelf: PropTypes.func.isRequired,
  shelf: PropTypes.string,
};

const defaultProps = {
  subtitle: '',
  description: '',
  authors: [],
  shelf: 'none',
};

const SELECTED_VALUE = 2;

const style = {
  bookCard: { heigth: 250, width: 210, padding: 3, margin: 10 },
  bookCover: { heigth: 250, width: 203, maxWidth: 203, minWidth: 203 },
};

class Book extends Component {
  constructor(props) {
    super(props);
    this.state = { shelf: props.shelf };
    this.onChangeBookShelf = this.onChangeBookShelf.bind(this);
  }

  onChangeBookShelf(...args) {
    this.setState(
      { shelf: args[SELECTED_VALUE] },
      this.props.onChangeBookShelf(args[SELECTED_VALUE])
    );
  }

  render() {
    const { title, authors, coverLink } = this.props;
    const { shelf } = this.state;

    return (
      <Card style={style.bookCard}>
        <CardMedia>
          <div className="cover-wrapper">
            <img src={coverLink} alt={title} style={style.bookCover} />
          </div>
        </CardMedia>
        <CardText>{authors.join(', ').trim()}</CardText>
        <CardActions>
          <DropDownMenu value={shelf} onChange={this.onChangeBookShelf}>
            <MenuItem value="none" primaryText="None" />
            <MenuItem value="currentlyReading" primaryText="Currently Reading" />
            <MenuItem value="wantToRead" primaryText="Want To Read" />
            <MenuItem value="read" primaryText="Read" />
          </DropDownMenu>
        </CardActions>
      </Card>
    );
  }
}

Book.propTypes = propTypes;
Book.defaultProps = defaultProps;

export default Book;
