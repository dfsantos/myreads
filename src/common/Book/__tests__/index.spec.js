import React from 'react';
import toJson from 'enzyme-to-json';

import Book from '../index';

const TITLE = "Book's Title";
const AUTHORS = ['Author of Book'];

it('should render without errors', () => {
  shallow(<Book title={TITLE} authors={AUTHORS} />);
});

it('default snapshot', () => {
  const wrapper = shallow(<Book title={TITLE} authors={AUTHORS} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
