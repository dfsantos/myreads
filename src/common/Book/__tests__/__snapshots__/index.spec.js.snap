// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default snapshot 1`] = `
<div
  className="book"
>
  <div
    className="book-top"
  >
    <div
      className="book-cover"
      style={
        Object {
          "backgroundImage": "url()",
          "height": 193,
          "width": 128,
        }
      }
    />
    <div
      className="book-shelf-changer"
    >
      <select
        defaultValue="none"
        onChange={[Function]}
      >
        <option
          disabled={true}
          value="none"
        >
          Move to...
        </option>
        <option
          value="currentlyReading"
        >
          Currently Reading
        </option>
        <option
          value="wantToRead"
        >
          Want to Read
        </option>
        <option
          value="read"
        >
          Read
        </option>
        <option
          value="none"
        >
          None
        </option>
      </select>
    </div>
  </div>
  <div
    className="book-title"
  >
    Book's Title
  </div>
  <div
    className="book-authors"
  />
</div>
`;
